lms/templates/problem.html

```diff
@@ -98,6 +101,7 @@ <h3 class="hd hd-2 problem-header" id="${ short_id }-problem-title" aria-describ
       notification_type='warning',
       notification_icon='fa-save',
       notification_name='save',
-      notification_message=''"
+      notification_message=save_message,
+      is_hidden=not has_saved_answers"
```

Comment:
Instead of adding this "is_hidden" property, can't you key off of whether or not a non-empty save_message exists? Then the server code (vs. template) is deciding what to render based on state.